### PR TITLE
`Development`: Fix running server tests on windows

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context
This PR fixes a few issues present when running the server tests locally on Windows.

### Description
- The method `extractZipFileRecursively` opened a ZipFile, but was not closing it, leading to issues when deleting the file later on. I also simplified the Method.
- `extractZipFileRecursively()` now directly returns the created folder to access its files later in the test.
- The `FileServiceTest` was using `URL.create()` after calling `toString()` on a `Path`. On Windows, Paths contain backslashes, but URIs can only contain forward slashes. We now replace backslashes with forward slashes.
- Fixes multiple occurrences of temporary files that got created but never deleted.
- Removes `@DisableOnOs` annotations and fixes unclosed test resources.
- Refactored and improved some places.

### Steps for Testing
code review
run the tests on Windows if you own a Windows setup.

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java [hunk 1]
```diff
@@ -315,21 +315,11 @@ void testNotifyUserAboutAssessedExerciseSubmission() {
...
 
     /**
-     * Test for checkNotificationForAssessmentExerciseSubmission method with an undefined release date
-     */
-    @Test
-    void testCheckNotificationForAssessmentExerciseSubmission_undefinedAssessmentDueDate() {
-        exercise = TextExerciseFactory.generateTextExercise(null, null, null, course);
-        singleUserNotificationService.checkNotificationForAssessmentExerciseSubmission(exercise, user, result);
-        verify(singleUserNotificationService).checkNotificationForAssessmentExerciseSubmission(exercise, user, result);
-    }
```
Comment (laadvo) [inline src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java:325]: Is there a reason you removed this test?

File: src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java [hunk 2]
```diff
@@ -315,21 +315,11 @@ void testNotifyUserAboutAssessedExerciseSubmission() {
...
-
-    /**
-     * Test for checkNotificationForExerciseRelease method with a current or past release date
+     * Test for checkNotificationForExerciseRelease method with a past assessment due date
      */
     @Test
     void testCheckNotificationForAssessmentExerciseSubmission_currentOrPastAssessmentDueDate() {
```
Comment (laadvo) [inline src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java:321]:
```suggestion
    void testCheckNotificationForAssessmentExerciseSubmission_pastAssessmentDueDate() {
```
Also, why did you change the assessment due date to `ZonedDateTime.now().minusMinutes(1)` ? I'm guessing it was flaky

**Review by laadvo — APPROVED** [review]
Code LGTM, I just left two small questions.
Thanks for fixing the tests on windows and removing all @`DisableOnOs` annotations :) 
The related tests still succeed on Bamboo and Github Actions. I don't use windows so I can't check them there.

File: src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java (hunk 2 above)
Comment (Strohgelaender) [reply src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java:321]: Yes, the production code checks `now.isAfter(assessementDueDate)` and the test was flaky if the same timestamp was now and the saved assessment due date.

File: src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java (hunk 1 above)
Comment (laadvo) [reply src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java]: You're right :+1:

**Pull request merged**